omegaconf/dictconfig.py

```diff
@@ -574,7 +575,8 @@ def _set_value(self, value: Any, flags: Optional[Dict[str, bool]] = None) -> Non
             self.__dict__["_content"] = "???"
             self._metadata.object_type = None
         else:
-            previous_state = copy.copy(self.__dict__["_content"])
+            # shallow copy of content
+            self.__dict__["_previous_content"] = self.__dict__["_content"]
```

Comment:
I think we should absolutely not go this way.
Set should be atomic, you are introducing a new object state change here.